src/ert/scheduler/job.py

```diff
@@ -66,36 +69,56 @@ async def __call__(
         self, start: asyncio.Event, sem: asyncio.BoundedSemaphore
     ) -> None:
         await start.wait()
-        await sem.acquire()
-
-        try:
-            await self._send(State.SUBMITTING)
-            await self.driver.submit(
-                self.real.iens, self.real.job_script, cwd=self.real.run_arg.runpath
-            )
-
-            await self._send(State.STARTING)
-            await self.started.wait()
-
-            await self._send(State.RUNNING)
-            returncode = await self.returncode
-            if (
-                returncode == 0
-                and forward_model_ok(self.real.run_arg).status
-                == LoadStatus.LOAD_SUCCESSFUL
-            ):
-                await self._send(State.COMPLETED)
-            else:
-                await self._send(State.FAILED)
-
-        except asyncio.CancelledError:
-            await self._send(State.ABORTING)
-            await self.driver.kill(self.iens)
-
-            await self.aborted.wait()
-            await self._send(State.ABORTED)
-        finally:
-            sem.release()
+        retries = 0
+        retry: bool = True
+        while retry:
+            retry = False
+            await sem.acquire()
+            try:
+                await self._send(State.SUBMITTING)
+                await self.driver.submit(
+                    self.real.iens, self.real.job_script, cwd=self.real.run_arg.runpath
+                )
+
+                await self._send(State.STARTING)
+                await self.started.wait()
+
+                await self._send(State.RUNNING)
+                returncode = await self.returncode
+                # we need to make sure that the task has finished too
+                await self.driver.wait(self.real.iens)
+
+                if (
+                    returncode == 0
+                    and forward_model_ok(self.real.run_arg).status
+                    == LoadStatus.LOAD_SUCCESSFUL
+                ):
+                    await self._send(State.COMPLETED)
+                else:
+                    await self._send(State.FAILED)
+                    retries += 1
+                    retry = retries < self._scheduler._max_submit
```

Comment:
The scheduler should probably propagate this max_submit down to the Job instance upon creation so we don't need to jump up to fetch it.